lib/ui/screens/home.dart

```diff
@@ -9,6 +12,13 @@ class HomePage extends StatefulWidget {
 }
 
 class _HomePageState extends State<HomePage> {
+  PersistencePort database = PersistenceAdapter();
```

Comment:
This gets recreated everytime i think. When you are adding firebase so actually data is fetched(which takes time and resources you should keep the Adapters/services outside the UI). By using Riverpod you can keep it all managed quite easily